src/FS.Embedded/Manifest/EmbeddedFilesManifest.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Extensions.FileProviders.Embedded.Manifest
+{
+    internal class EmbeddedFilesManifest
+    {
+        private static readonly char[] _invalidFileNameChars = Path.GetInvalidFileNameChars()
+            .Where(c => c != Path.DirectorySeparatorChar && c != Path.AltDirectorySeparatorChar).ToArray();
+
+        private static readonly char[] Separators = new char[] { Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar };
+        private readonly ManifestDirectory _rootDirectory;
+
+        internal EmbeddedFilesManifest(ManifestDirectory rootDirectory)
+        {
+            if (rootDirectory == null)
+            {
+                throw new ArgumentNullException(nameof(rootDirectory));
+            }
+
+            _rootDirectory = rootDirectory;
+        }
+
+        internal ManifestEntry ResolveEntry(string path)
+        {
+            if (string.IsNullOrEmpty(path) || HasInvalidPathChars(path))
+            {
+                return null;
```

Comment:
You should throw instead.

Reply:
Null signals a malformed path. The general guideline with these types is to use a special value instead of throwing:
See:
https://github.com/aspnet/FileSystem/pull/310/files#diff-498192ac768bf12b9b2d56daea3ff201R72
https://github.com/aspnet/FileSystem/pull/310/files#diff-498192ac768bf12b9b2d56daea3ff201R91